pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -78,6 +84,12 @@ abstract class Analytics {
     // resolve on their own
     const FileSystem fs = LocalFileSystem();
 
+    // Ensure that the home directory has permissions enabled to write
+    final Directory homeDirectory = getHomeDirectory(fs);
```

Comment:
This is unrelated to this change, but it looks to me like the user will get a null check error on line 103 (of the new diff) if they don't have a `HOME` env var set. I think we should handle that situation (it's rare, but happens).

Reply:
That's a good catch, `getHomeDirectory()` is now nullable so if we detect `homeDirectory` is null `or` we don't have write permissions, then we will use `NoOpAnalytics`

Reply:
cool!